src/track/macroaction.cpp

```diff
@@ -0,0 +1,22 @@
+#include "macroaction.h"
+
+proto::Macro_Action* MacroAction::serialize() const {
+    auto* serialized = new proto::Macro_Action();
```

Comment:
Why is proto::Macro_Action() allocated dynamically? Returning a plain pointer is deprecated. Either return it by value or use std::unique_ptr

Reply:
isn't allocating a `unique_ptr` overkill here as there is not much passing around?

Reply:
Doesn't matter whether you are passing it around or not. a `unique_ptr` does not have much more overhead than the plain allocation already has. Compared to an owning raw pointer it does come with some essential safety guarantees.

Reply:
using a std::unique_ptr produces a segfault, I actually need the raw pointer in this case

Reply:
Where does it produce a segfault, how are you creating and using the unique_ptr? You've used `make_unique<>()` right?
As a rule of thumb, there is no reason to use a raw pointer in modern C++ unless you need it for interacting with some C or legacy API.

Reply:
Just looked at your commit, you misused unique_ptr (welcome to C++ I guess). Take a look at this article from MS on how to use it: https://docs.microsoft.com/en-us/cpp/cpp/how-to-create-and-use-unique-ptr-instances?view=msvc-170

Reply:
std::unqiue_ptr is designed for moving dynamically allocated references when used correctly. The segfault will be caused by wrong usage. Please understand move semantics and how std::unique_ptr is supposed to work.

But we don't even need to allocate dynamically here! Please return by value and use `Add()` instead of `AddAllocated()` on the caller side.

Reply:
Doesn't passing by value copy the result?

Reply:
https://en.cppreference.com/w/cpp/language/copy_elision

Reply:
Ah right. Somehow the Ubuntu build fails after that change though, is there something wrong with how I did it?

Reply:
@Swiftb0y @uklotzde

Reply:
From what I can tell, you did nothing wrong, it seems to be an issue with how protobufs are implemented. It's weird and I have 0 idea why that is happening. I don't have the time capacity to look into the issue myself.